src/pymor/bindings/fenics.py

```diff
@@ -138,10 +138,7 @@ class FenicsVectorSpace(ComplexifiedListVectorSpace):
 
     def __init__(self, V, id='STATE'):
         self.__auto_init(locals())
-
-    @property
-    def dim(self):
-        return df.Function(self.V).vector().size()
+        self.dim = df.Function(self.V).vector().size()
```

Comment:
This means that creating a new `FenicsVectorSpace` always entails creating a new `Function` object and, thus, allocating memory for a DOF vector that is never used. I generally assume that creating a `VectorSpace` is cheap.
Note that there are very few places in pyMOR where the dimension of a `VectorSpace` actually is relevant.

(The current code is not optimal either, of course. The dimension should be cached.)